modules/olm-bundle-format-dependencies-file.adoc

```diff
@@ -8,10 +8,11 @@
 
 The dependencies of an Operator are listed in a `dependencies.yaml` file in the `metadata/` folder of a bundle. This file is optional and currently only used to specify explicit Operator-version dependencies.
 
-The dependency list contains a `type` field for each item to specify what kind of dependency this is. There are two supported types of Operator dependencies:
+The dependency list contains a `type` field for each item to specify what kind of dependency this is. There are three supported types of Operator dependencies:
 
 * `olm.package`: This type indicates a dependency for a specific Operator version. The dependency information must include the package name and the version of the package in semver format. For example, you can specify an exact version such as `0.5.2` or a range of versions such as `>0.5.1`.
 * `olm.gvk`: With a `gvk` type, the author can specify a dependency with group/version/kind (GVK) information, similar to existing CRD and API-based usage in a CSV. This is a path to enable Operator authors to consolidate all dependencies, API or explicit versions, to be in the same place.
+* `olm.constraint`: It declares a dependency constraint of a particular type, differentiating non-constraint and constraint properties. Its `value` field is an object containing a `failureMessage` field holding a string-representation of the constraint message that will be surfaced to the users if the constraint is not satisfiable at runtime.
```

Comment:
I would like to add a bit more into this paragraph given it is a bit vague. Can you take a look at the upstream doc and see what you can gather from there to improve this paragraph a bit? Plus, we need to include an example on `olm.constraint` in the example yaml below as well. There are some examples in the upstream doc that you can see as well.
https://olm.operatorframework.io/docs/concepts/olm-architecture/dependency-resolution/

Reply:
@dinhxuanvu Thanks for the feedback.
As we are referring content from an upstream doc, I believe we can provide a link to it, to avoid content duplication and future updates doc work.

My suggestion:

> `olm.constraint` - It declares a generic constraint on arbitrary operator properties. For more information, refer to [Generic Constraint Dependencies](https://olm.operatorframework.io/docs/concepts/olm-architecture/dependency-resolution/#generic-constraint).

Reply:
@dinhxuanvu and @Xia-Zhao-rh Requested fix has already been submitted by PR https://github.com/openshift/openshift-docs/pull/42318.

Doc Preview Link: https://deploy-preview-42318--osdocs.netlify.app/openshift-enterprise/latest/operators/understanding/olm/olm-understanding-dependency-resolution.html#olm-dependencies_olm-understanding-dependency-resolution

Let me know if I can close this PR. Thanks.